src/pages/General.vue

```diff
@@ -40,8 +41,7 @@ export default {
   components: { TelemetryOptIn, UpdateStatus },
   data() {
     return {
-      /** @type Settings */
-      settings:    ipcRenderer.sendSync('settings-read'),
+      settings:    settingsDefault.defaultTestConfig,
```

Comment:
Exported the default settings from `settings.ts` and passed it into the data(). It had fixed the error `Cannot read property 'type' of null` and the bug on the checkbox.
Unfortunately it had introduced another issue (see image attached). Invoking the app using `jest`, `mocha` or `playwright`, crashes the version check method.
![Screen Shot 2021-10-29 at 12 48 53 PM](https://user-images.githubusercontent.com/37411123/139495523-c22163c5-709d-4680-b4d0-3b34fec35bea.png)
![Screen Shot 2021-10-29 at 12 49 11 PM](https://user-images.githubusercontent.com/37411123/139495548-2ebc9c53-18ed-4bad-bf4b-78d7e5c7e4ed.png)
When it happens, `jest` just hangs and throw an error: `"javascript error: Cannot read property 'type' of null\n  (Session info: chrome=89.0.4389.128)"`... 

The interesting thing is: This error shows up during the `app.start()` call and seems to be isolated on `chromedriver`. It's a black box - didn't find a good way to place some breakpoints...
Going a bit deeper, here's the loading process:
1. `jest` will launch the spec file
2. `electron pre-config` will send the customs args to main app
3. `app.start()` will invoke the chromedriver.js
4. `chromediver` will invoke `launcher.js` and it will start the `browser_init`
5. `browser_init` will open the app and crash the app version check. 

Also, this is an intermittent issue (maybe another racing condition?) :( 
Running out of ideas...
Do you guys have any thoughts/ideas? Let's brainstorming
@mook-as @ericpromislow @rak-phillip

Reply:
There's a potential suspect for this issue and it could be related to the Electron/Spectron version.
We didn't find which step the `type` is receiving `null` (maybe `process.type` from main/renderer process).

We have decided to move the fix on small PRs:
* Updater fix - https://github.com/rancher-sandbox/rancher-desktop/pull/855
* Type of null (Electron upgrade)- Open a new PR